hypothesis-python/src/hypothesis/internal/validation.py

```diff
@@ -113,11 +113,18 @@ def check_valid_size(value, name):
                 'min_size=None is deprecated; use min_size=0 instead.'
             )
         return
-    check_type(integer_types + (float,), value)
-    if value < 0:
-        raise InvalidArgument(u'Invalid size %s=%r < 0' % (name, value))
     if isinstance(value, float) and math.isnan(value):
```

Comment:
It's more efficient in the non-deprecated case to write

```python
if isinstance(value, float):
    if math.isnan(value):
        raise ...
    from ... import ...
```

so we only ever check whether the value is a float once.